Module 2

GeorgiiDermenzhy opened the pull request:
In this pull request I've added ...

Task_ 2 of Module_2

# Pull Request Checklist

<!-- Please keep this section. It will make the maintainer's life easier. -->

* [x] I've checked code quality with flake8 and pylint tools
* [x] I've checked compliance with Python docstring conventions with pydocstyle tool
* [x] All third party packages are added to requirements.txt

GeorgiiDermenzhy committed:
Complete Task 1 of Module 2

GeorgiiDermenzhy committed:
Add task 2 from Module 1

GeorgiiDermenzhy committed:
PULL_REQUEST_TEAMPLATE file update

Add information on task and modify checkboxes

GeorgiiDermenzhy committed:
Fix PULL_REQUEST_TEMPLATE FILE

The comment removed and checkboxes unchecked

GeorgiiDermenzhy committed:
Modify task_2 of Module 2 with:

create CONSTANT variables for weight and implemented in display_inventory function
update dragonLoot variable changed to dragon_loot
change header for display_inventory function
change print iteams , to use default print() separator settings
change item_total counter to use augmented assignment
here notify our hero if he/she is carrying too much
add implementation of defaultdict for skipped dict
change print for number of added items, to use f-strings

GeorgiiDermenzhy committed:
Merge branch 'module_2' of https://github.com/GeorgiiDermenzhy/Basics-of-Python into module_2

GeorgiiDermenzhy committed:
Modify task_2 of Module 2 with below and checked with pydocstyle

create CONSTANT variables for weight and implemented in display_inventory function
update dragonLoot variable changed to dragon_loot
change header for display_inventory function
change print iteams , to use default print() separator settings
change item_total counter to use augmented assignment
here notify our hero if he/she is carrying too much
add implementation of defaultdict for skipped dict
change print for number of added items, to use f-strings

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+"""Task 1 and 2 of Module 2."""
+
+from collections import defaultdict
+# create CONSTANT variables for weight
```
Those notes look more like self TODO notes, not sure if we need them in the final version? :)

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+stuff = {'rope': 1, 'torch': 6, 'gold coin': 42, 'dagger': 1, 'arrow': 12}
+# update dragonLoot variable changed to dragon_loot
+dragon_loot = [
+
```
This extra space is not necessary, could you scan the code with pylint/flake8 and try to address the styling issues?

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+def display_inventory(inventory):
+    # change header for display_inventory function
+    """Check the inventory of a player and provides weight/speed dependency status."""
+    print("Inventory:")
```
I'd keep this print statement directly over the for loop from line 24 to keep it nice and close to each other

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+
+    print("Total number of items: " + str(item_total))
+    # here notify our hero if he/she is carrying too much
+    if item_total >= LIGHT_WEIGHT_THRESHOLD:
```
I think the logic here should be the other way around. We should start with the heaviest and go to the lowest weight. If `item_total` will exceed `MAX_INVENTORY_CAPACITY`, then this statement will be fired, as opposed to statement from line 40.

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+
+stuff = add_to_inventory(stuff, dragon_loot)
+
+display_inventory(stuff)
```
Missing new line at the end of the file

Pacu2 commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,81 @@
+    added_items_counter = 0
+
+    # your code goes here
+    trash = ['rubbish', 'chewed gum', 'used tissue']
```
It's more of `unwanted_items`, you know what they're saying, one man's trash is another man's treasure.

This argument is hardcoded but doesn't change between function runs. What would you say about introducing a CONST and moving it to the top of the file, out of function's scope?

GeorgiiDermenzhy committed:
Make changes based on pull request comments:
1) create UNWANTED_ITEMS const
2) fix weight CONST place to start check with heaviest
3) remove comments

GeorgiiDermenzhy committed:
Remove spaces in dragon_loot list

musk-ox commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,79 @@
+
+        print(item_name, item_count)
+
+        item_total += 1
```
I think it should be `item_total += item_count`. Currently it says:
```
Inventory:
rope 1
torch 6
gold coin 45
dagger 2
arrow 12
ruby 1
Total number of items: 6
```
so it counts types of items instead of total number of all items.

musk-ox commented on tasks/Module 2/task_1.py:
```diff
@@ -0,0 +1,79 @@
+    print("Inventory:")
+
+    for item_name, item_count in inventory.items():
+
```
I would rather keep it compact and remove this empty line. And maybe empty line after print.

GeorgiiDermenzhy committed:
Remove spaces and update total_items counter

GeorgiiDermenzhy committed:
Add new line at the end of the file

GeorgiiDermenzhy committed:
Remove unused variable "skippedd = {}"
